core/src/main/scala/kafka/coordinator/group/GroupMetadata.scala

```diff
@@ -244,6 +244,38 @@ private[group] class GroupMetadata(val groupId: String, initialState: GroupState
       leaderId = members.keys.headOption
   }
 
+  /**
+    * Check whether current leader is rejoined. If not, try to find another joined member to be
+    * new leader. Return false if
+    *   1. the group is currently empty (has no designated leader)
+    *   2. no member rejoined
+    */
+  def maybeElectNewJoinedLeader(): Boolean = {
+    leaderId.fold (false) { currentLeaderId => {
```

Comment:
nit: this use of `fold` is a tad odd. Perhaps we could use `leaderId.map{ ... }.getOrElse(false)`? Also, we don't need the second braces.